lib/ramble/ramble/reports.py

```diff
@@ -326,6 +330,23 @@ def add_minmax_data(self, selected_data, min_data, max_data, scale_var):
                 from_col=ReportVars.FOM_VALUE_MAX.value,
             )
 
+    def add_to_inventory(self, filename):
+        """Adds a filename to the inventory.
+
+        Args:
+            filename: filename to add to inventory.
+        """
+        try:
+            with open(self.inventory_path) as f:
+                inventory = syaml.load(f)
+        except FileNotFoundError:
+            inventory = {"files": []}
+
+        inventory["files"].append(filename)
+
+        with open(self.inventory_path, "w+") as f:
```

Comment:
Instead of reading and writing the inventory file for each file we write, can we batch the updates to the inventory file?

Doing this with a large report (i.e. lots of png / pdf files) could cause a lot of thrash on a shared filesystem.

Reply:
I've updated it to only write once per inventory file, and I think it can skip the 'read' and just initialize a new empty inventory because each new plot generator / reports folder has a unique timestamped path created on invocation.

Reply:
Awesome, thanks!